src/Features/Core/Portable/RemoveUnusedExpressionsAndParameters/AbstractRemoveUnusedExpressionsAndParametersDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,755 @@
+                                                         additionalLocations: null,
+                                                         properties: null,
+                                                         arg1,
+                                                         arg2);
+                reportDiagnostic(diagnostic);
+            }
+
+            private static IEnumerable<INamedTypeSymbol> GetAttributesForMethodsToIgnore(Compilation compilation)
+            {
+                // Ignore conditional methods (One conditional will often call another conditional method as its only use of a parameter)
+                var conditionalAttribte = compilation.ConditionalAttribute();
+                if (conditionalAttribte != null)
+                {
+                    yield return conditionalAttribte;
+                }
+
+                // Ignore methods with special serialization attributes (All serialization methods need to take 'StreamingContext')
+                var onDeserializingAttribute = compilation.OnDeserializingAttribute();
+                if (onDeserializingAttribute != null)
+                {
+                    yield return onDeserializingAttribute;
+                }
+
+                var onDeserializedAttribute = compilation.OnDeserializedAttribute();
+                if (onDeserializedAttribute != null)
+                {
+                    yield return onDeserializedAttribute;
+                }
+
+                var onSerializingAttribute = compilation.OnSerializingAttribute();
+                if (onSerializingAttribute != null)
+                {
+                    yield return onSerializingAttribute;
+                }
+
+                var onSerializedAttribute = compilation.OnSerializedAttribute();
+                if (onSerializedAttribute != null)
+                {
+                    yield return onSerializedAttribute;
+                }
+
+                // Don't flag obsolete methods.
+                var obsoleteAttribute = compilation.ObsoleteAttribute();
+                if (obsoleteAttribute != null)
+                {
+                    yield return obsoleteAttribute;
+                }
+            }
+
+            private bool IsUnusedParameterCandidate(IParameterSymbol parameter)
+            {
+                // Ignore implicitly declared parameters and methods, extern methods, abstract methods,
+                // virtual methods, overrides, interface implementations, accessors and lambdas.
+                if (parameter.IsImplicitlyDeclared ||
+                    parameter.Name == "_" ||
+                    !(parameter.ContainingSymbol is IMethodSymbol method) ||
+                    method.IsImplicitlyDeclared ||
+                    method.IsExtern ||
+                    method.IsAbstract ||
+                    method.IsVirtual ||
+                    method.IsOverride ||
+                    !method.ExplicitOrImplicitInterfaceImplementations().IsEmpty ||
```

Comment:
i don't understand how ExplicitOrImplicitInterfaceImplementations could possibly work.  It has no concept of hte great solutoin (or what lies outside the solution) so it can't know if a member is implementing an interface member for someone.

Reply:
This works at compilation level, so it will not catch the case you mentioned, but I would argue it is not really a common enough pattern to force us to not fire for all non-private methods, as majority of such cases are truly redundant parameters. If everyone feels strongly, we can improve the message to indicate this possibility - note there is no code fix here.